storagemarket/provider.go

```diff
@@ -451,6 +451,11 @@ func (p *Provider) Start() error {
 			continue
 		}
 
+		// Fail deals if start epoch has passed
+		if err := p.checkDealProposalStartEpoch(deal); err != nil {
+			p.failDeal(dh.Publisher, deal, err, false)
+		}
+
```

Comment:
I think there's a continue statement missing here.

Also failDeal should be called inside a go-routine. It will send a message to the main event loop, and that will block indefinitely if it's not called in a go-routine (because at this point the event loop hasn't started yet).

If possible we should try to add a test to catch these kinds of bugs.